### ISSUE TITLE
Cannot resolve `await import('https://cdn.jsdelivr.net/npm/generate-ip/dist/generate-ip.min.js');`

### ISSUE THREAD

**Issue by adamlui** [import-js/eslint-plugin-import#3118]
https://github.com/adamlui/js-utils/blob/23b8bbb60944f8207e9793ad3ec021ffedb39f9e/geolocate/src/geolocate.js#L23

```js
            await import('https://cdn.jsdelivr.net/npm/generate-ip/dist/generate-ip.min.js');
```

results in `error  Unable to resolve path to module 'https://cdn.jsdelivr.net/npm/generate-ip/dist/generate-ip.min.js'  import/no-unresolved`

**Comment by michaelfaith** [comment]
Duplicate of https://github.com/import-js/eslint-plugin-import/issues/3107

**Comment by adamlui** [comment]
Actually I tested using a random npm pkg with unminified js and same false positive

**Comment by michaelfaith** [comment]
Can you try running with `DEBUG=eslint-plugin-import:*` and sharing the output?

**Comment by adamlui** [comment]
Sure

```
  eslint-plugin-import:resolver:node Resolving: https://cdn.jsdelivr.net/npm/generate-ip/dist/generate-ip.min.js from: e:\js\utils\geolocate\src\geolocate.js +1s
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.jsdelivr.net/npm/generate-ip/dist/generate-ip.min.js' from 'e:\js\utils\geolocate\src'
    at resolveSync (e:\js\utils\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\js\utils\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\js\utils\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\js\utils\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\js\utils\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\js\utils\node_modules\eslint-module-utils\resolve.js:233:12)
    at checkSourceValue (e:\js\utils\node_modules\eslint-plugin-import\lib\rules\no-unresolved.js:40:53)
    at checkSourceValue (e:\js\utils\node_modules\eslint-module-utils\moduleVisitor.js:32:5)
    at checkImportCall (e:\js\utils\node_modules\eslint-module-utils\moduleVisitor.js:62:5)
    at ruleErrorHandler (e:\js\utils\node_modules\eslint\lib\linter\linter.js:1098:48) {
  code: 'MODULE_NOT_FOUND'
} +2ms
```

**Comment by adamlui** [comment]
Also it happens using classic syntax too w/ the URL @ https://github.com/KudoAI/kudoai.com/blob/78a93d41671059713d26d081bb5a00d368311a1f/assets/js/matrix-bg.js#L1

```js
import * as datGui from 'https://cdn.skypack.dev/dat.gui@0.7.7'
```

**Comment by adamlui** [comment]
For that one it strangely tries multiple times

```
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +127ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at Function.get (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\exportMap\builder.js:37:45)
    at processBodyStatement (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\namespace.js:12:38)
    at e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\namespace.js:85:40
    at Array.forEach (<anonymous>) {
  code: 'MODULE_NOT_FOUND'
} +4ms
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +7ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at checkSourceValue (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\no-unresolved.js:40:53)
    at checkSourceValue (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\moduleVisitor.js:32:5)
    at checkSource (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\moduleVisitor.js:38:5)
    at ruleErrorHandler (e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\linter.js:1098:48) {
  code: 'MODULE_NOT_FOUND'
} +2ms
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +3ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at Function.get (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\exportMap\builder.js:37:45)
    at checkDefault (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\default.js:22:44)
    at ruleErrorHandler (e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\linter.js:1098:48)
    at e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\safe-emitter.js:45:58 {
  code: 'MODULE_NOT_FOUND'
} +1ms
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +2ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at defaultResolver (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\no-duplicates.js:320:112)
    at ImportDeclaration (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\no-duplicates.js:359:32)
    at ruleErrorHandler (e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\linter.js:1098:48)
    at e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\safe-emitter.js:45:58 {
  code: 'MODULE_NOT_FOUND'
} +10ms
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +3ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at Function.get (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\exportMap\builder.js:37:45)
    at checkDefault (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\no-named-as-default.js:29:51)
    at ruleErrorHandler (e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\linter.js:1098:48)
    at e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\safe-emitter.js:45:58 {
  code: 'MODULE_NOT_FOUND'
} +2ms
  eslint-plugin-import:resolver:node Resolving: https://cdn.skypack.dev/dat.gui@0.7.7 from: e:\kudoai\kudoai.com\assets\js\matrix-bg.js +1ms
  eslint-plugin-import:resolver:node resolve threw error: Error: Cannot find module 'https://cdn.skypack.dev/dat.gui@0.7.7' from 'e:\kudoai\kudoai.com\assets\js'
    at resolveSync (e:\kudoai\kudoai.com\node_modules\resolve\lib\sync.js:111:15)
    at exports.resolve (e:\kudoai\kudoai.com\node_modules\eslint-import-resolver-node\index.js:22:20)
    at withResolver (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:180:23)
    at fullResolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:201:22)
    at relative (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:217:10)
    at resolve (e:\kudoai\kudoai.com\node_modules\eslint-module-utils\resolve.js:233:12)
    at Function.get (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\exportMap\builder.js:37:45)
    at ImportDefaultSpecifier (e:\kudoai\kudoai.com\node_modules\eslint-plugin-import\lib\rules\no-named-as-default-member.js:39:50)
    at ruleErrorHandler (e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\linter.js:1098:48)
    at e:\kudoai\kudoai.com\node_modules\eslint\lib\linter\safe-emitter.js:45:58 {
  code: 'MODULE_NOT_FOUND'
} +2ms

e:\kudoai\kudoai.com\assets\js\matrix-bg.js
  1:20  error  Unable to resolve path to module 'https://cdn.skypack.dev/dat.gui@0.7.7'  import/no-unresolved
```

**Comment by adamlui** [comment]
> This isn’t surprising because node doesn’t support importing URLs. To resolve something with a URL, you’d need a custom resolver.

So your plugin being marketed as "help validate proper imports" and https://github.com/KudoAI/kudoai.com/blob/78a93d41671059713d26d081bb5a00d368311a1f/assets/js/matrix-bg.js#L1

```js
import * as datGui from 'https://cdn.skypack.dev/dat.gui@0.7.7'
```

...is valid JS import (properly generates matrix code background @ https://kudoai.com) means plugin should be expected to also lint URL imports regardless of what node supports right, or are you only targeting node users

(also I intend to convert my other pkgs like chatgpt.js to be node compatible eventually, then there will be `import(url)` still in codebase for frontend devs that will still be linted so the false positives will persist even if I myself pivot to targeting node)

> (also, don’t use CDNs; always host your own assets)

I'm not rich sir <3 free jsDelivr (but I kinda agree when they go down my so many users are rekt)

**Comment by ljharb** [comment]
No, it’s not a valid JS import, it’s a valid browser import, which we don’t support.

**Comment by adamlui** [comment]
Browser is not a language though, it is JS syntax (ES6) for 9 years and correctly resolved the resource so is valid JavaScript import and ESLint was made to lint JS (regardless of env) so I thought your plugin enhances the linter you're plugging into thus adheres to same principle of supporting all valid JS , but you have confirmed I am wrong (which is why I asked are you targeting node users only, again browser users also write in valid JS, but now I know the answer)

**Comment by adamlui** [comment]
If it's valid syntax as you admit then it's valid JS import (JS literally means JavaScript, not what specifier is passed) which **all ESLint plugins** lint (regardless of env) except for yours.

The spec not discussing specifier (syntax is ES6 btw not ECMA-262, valid for 9 years already as previously stated, much older than you're falsely suggesting 1 year) does not change this fact valid JS is being written. ESLint (+ JSLint) and its plugins effectively analyze JS regardless of the env. Yours simply doesn't.

Also you should reflect on your crude labeling of JS in Node.js as "JS import" but JS in browser as not. They're **both written in JS**, you can't call one not. Before Node.js ever gained popularity JS was being written since the 90s, specifically for browsers. You can even pass relative paths as specifiers in **_both_** envs using the exact same valid JS syntax, so it's absurd to only call one of them JavaScript

**Comment by ljharb** [comment]
The contents of the string specifier aren’t part of JS, no. The syntax you’re using of course is, but that’s not where your problem lies.

You’re asking about resolving a URL, which is browser-specific territory and not governed by the JS spec whatsoever. I didn’t say anything about a year; I’m very aware how the spec works having been on the spec committee for a decade.

I understand that you’re upset, but *this project simply does not resolve URLs*, and never intended nor intends to, and if you want that feature, you’ll have to find your own custom import resolver (which is discussed in the readme).

**Comment by ljharb** [comment]
“all valid JS” is impossible to target, because there’s infinite possibilities. Supporting URLs is *just browsers* (and deno, i suppose).

**Comment by ljharb** [comment]
Nobody said 1 year? Ecma-262 is the standard that comes out with a new edition every year, and is an umbrella term for “every version of JS ever”.

**Comment by adamlui** [comment]
> Nobody said 1 year? Ecma-262 is the standard that comes out with a new edition every year, and is an umbrella term for “every version of JS ever”.

Sorry DDG showed this

![Image](https://github.com/user-attachments/assets/4fa407f6-09a7-43db-9e52-aeb8c3a5cd4e)

**Comment by adamlui** [comment]
> “all valid JS” is impossible to target, because there’s infinite possibilities. Supporting URLs is _just browsers_ (and deno, i suppose).

But you dismiss it as "just" browsers even after acknowledging JS for browsers preceded Node.js, so why ignore such a large portion of valid JS?

**Comment by adamlui** [comment]
i.e. if you install any random ESLint plugin it will work with browser JS (of which alot of code is) so one would expect the same for yours for carrying the 'eslint-' prefix.

You earlier cited WHATWG HTML spec but I didn't get why, are you saying that makes it difficult for you to support browser JS, or that it's still easy but the spec made you decide browser JS (of which alot of code is) is somehow undeserving of same treatment

**Comment by adamlui** [comment]
Or you meant just because the specifier wasn't in ES specs, the whole statement (valid in browser JS, of which alot of code is and ESLint plugins typically support) should not be supported, if so how ccome

**Comment by ljharb** [comment]
`import` did not precede node.js; so the fact that browsers are older is pretty irrelevant.

I'm saying that the only place "importing URLs" is specified is in the browser-only WHATWG HTML spec. So yes, URLs are *simply not part of the JS language*. They're part of some JS *engines* which is an entirely distinct thing.

If you read the readme, you should be able to understand that this plugin doesn't include resolution rules, it uses one or more "resolvers". You can use the node resolver, or the webpack resolver; there's a typescript resolver, and potentially even a URL resolver - in other words, it's not this plugin's job to resolve URLs, it's an import resolver's job to do so.

**Comment by adamlui** [comment]
> `import` did not precede node.js; so the fact that browsers are older is pretty irrelevant.

It is highly relevant in that ESLint (and its plugins) lint JS to make it better agnostically of env (i.e. including/especially pre-node code) and you also carry the banner in the plugin's name, so are creating that expectation of lint JS (including/especially pre-node code) just like ESLint and its plugins 

> I'm saying that the only place "importing URLs" is specified is in the browser-only WHATWG HTML spec. So yes, URLs are simply not part of the JS language. They're part of some JS engines which is an entirely distinct thing.

So what? The specifier varying by JS env does not change the fact valid JS was used by authors who look to ESLint (and plugins) to improve code. That's like saying browsers support `console.log(document.body)` but node doesn't, so my plugin for ESLint the linter that's supposed to lint all JS will give error on `console.log(document.body)` but not some string that both envs support, just because document.body doesn't exist in node (doesn't change the fact `console.log()` is valid so should be linted w/o false positive)

Even earlier you strangely stated "This isn’t surprising because node doesn’t support importing URLs" which actually revealed to me it was likely intentional you don't treat imports outside node as deserving of ESLinting.

Using same analogy, that is equivalent to someone revealing `console.log(document.body)` false positive, then you make similar proclamations like "it is not valid JS log" to justify the false error. How do you not get that is just weird logic? The hypothetical scripter just wanted to enhance his ESLinting and expects `console.log(anything)` to _just work_ because it is truly valid JS and that is ESLint's job (validate truly valid JS). Your plugin claims to validate imports as an ESLint plugin so it should _just work_ with all valid JS like all ESLint plugins

> If you read the readme, you should be able to understand that this plugin doesn't include resolution rules, it uses one or more "resolvers". You can use the node resolver, or the webpack resolver; there's a typescript resolver, and potentially even a URL resolver - in other words, it's not this plugin's job to resolve URLs, it's an import resolver's job to do so.

But your plugin is called "eslint-plugin-import" whose naming convention implies lint the word after "eslint-plugin-" (which is **imports**), so if it can't resolve URLs in **imports** then it's failing to perform its job (not resolver's job) of "help validate proper imports"

And from your earlier mislabeling of this 100% valid and proper import as somehow "not proper JS import" I understand why you actively chose not to support it, all I'm trying to get you to see it's actually very proper/valid JS import because it is valid JS and resolves a bonus resource of URLs in a specific env supported by probably every ESLint plugin and yours is called eslint-plugin-something so your your description is misleading unless you change it to "some envs" or implement URL resolver out-the-box

**Comment by adamlui** [comment]
Also it will probably false positive even not URLs but just any relative path when used in browser but not in node, despite it actually resolved in both envs, so your logic one is somehow proper but the other is not is very strange (the code is identical) reason to not support one but do for other

**Comment by michaelfaith** [comment]
> But your plugin is called "eslint-plugin-import" not "eslint-plugin-node-import" whose naming convention implies lint the word after "eslint-plugin-" (which is **imports** not node imports), so if it can't resolve URLs in all **imports** then it's failing to perform its described job (not resolver's job) of "help validate proper imports" (not "help validate proper node imports" as you wote the former and named it ending in -imports)

You completely missed his point here.  He's explaining to you that the plugin is intended to be flexible enough support a wide range of resolution algorithms.  Those resolution algorithms are encapsulated as "import resolvers".  So you can plug and play different resolvers with this plugin to achieve different results.  There's the default node resolver, there's a typescript resolver, etc.  So if you really want the behavior that you're so passionately and endlessly arguing about, you're able to build a resolver that meets your needs and still use this plugin.  But it sounds like you're not super interested in solving the problem yourself, but instead want to argue as to why someone else should solve it for you.

**Comment by adamlui** [comment]
I don't think you understood his though, he's declared he intentionally will not support (out-the-box) URL resolving ("does not resolve URLs" were his words) because it is not proper JS (also his words) all because because URLs are spec'd by a different committee. **Even though import is proper JS** and all ESLint plugins (out-the-box) lint proper JS in browser env.

So this above declared intention is based on a very strange/irrelevant reason to be unlike all other plugins that carry the 'eslint-plugin-' banner (lint valid JS, including/especially in browser env, not force customer to build their own as you two strangely suggest)

**Comment by adamlui** [comment]
I already said I might build my own plugin to not dump the responsibility on users after it was revealed he is targeting node users, so your point was not lost on me btw (my point is it is very odd to ask users to do that just to make an ESLint plugin they don't even maintain support browsers)

**Comment by adamlui** [comment]
Also I don't think you realize just how significant the portion of ESLint (or any linter) users are beginners trying to improve their code. Asking a novice (not me but example) just getting into writing JS for browsers to write a complex URL resolver instead of just support them or update misleading description w/ "for node.js" or "some envs" is very unusual ask

**Comment by ljharb** [comment]
Beginners shouldn't be using URLs - they should be using a bundler.

**Comment by adamlui** [comment]
Beginners exist who aren't building apps but just writing scripts to hack a page (and are using ESLint to help) that's not necessarily even theirs (this is the historical use case of JS and is very proper, just like not using bundlers)

**Comment by ljharb** [comment]
That’s precisely eslint's job - it’s a linter. Also, such beginners won’t be using eslint.

**Comment by adamlui** [comment]
> That’s precisely eslint's job - it’s a linter

Linters literally lint code not env or build setups

![Image](https://github.com/user-attachments/assets/b1cd38e4-96e7-4652-a326-c30a6867513f)

> Also, such beginners won’t be using eslint

How come?

**Comment by adamlui** [comment]
Did you know beginners use linters because they're not as good as you and me yet so need to catch code errors

**Comment by ljharb** [comment]
I’m really not sure why this continues to be an argument.

This project’s job is to be opinionated and tell you that some things are good or bad. If you don’t like the recommendations, you can either disable them or not use the project.

**Comment by adamlui** [comment]
> I’m really not sure why this continues to be an argument.

Well you keep saying some really outlandish things so I simply reply to your notifications, if you're upset you could block me (closing the issue prematurely didn't seem to make me go away)

> This project’s job is to be opinionated and tell you that some things are good or bad

Indeed and for ESLint plugins, these "things" are code (not env or build setup)

> If you don’t like the recommendations, you can either disable them or not use the project

What recommendations

**Comment by adamlui** [comment]
Ooh the false positives? I did disable them I told you already (any continued discussion are my replies to notifications you keep sending me)

**Comment by adamlui** [comment]
Do you want me to stop typing sir?

**Comment by adamlui** [comment]
Also I did not view this as some unproductive argument but hopefully an eye-opening interaction to elucidate why your fundamental reasoning to alienate browser users + beginners goes against what ESLint is used for (lint code not build-setup and be env agnostic) is flawed. In the end it is your project, I'm not your boss, you could alienate who you please and it wouldn't affect me (I enjoy interaction though and replying to notifs but if you say the word I will stop typing to you)

**Comment by ljharb** [comment]
Importing a URL is improper and will slow down your application, and i hope beginners do learn that. Use a bundler.

**Comment by adamlui** [comment]
But not everyone is building an app so bundling is not proper, do you understand ESLint is a linter for code not just apps?

**Comment by adamlui** [comment]
I'll try a different tact, what was the first line of JavaScript you wrote for. Was it an app?

**Comment by ljharb** [comment]
History is irrelevant. Modern web dev requires using a bundler.

**Comment by adamlui** [comment]
Can you just answer the question? You keep being evasive. If the answer is it was not an app, would you have found use for a linter anyway? If so, wasn't the reason because you wanted proper code for this unbundled JS regardles of it's an app or not? If yes, then why are you demanding this past you to use a bundler for a case where it is infeasible/improper to do so, lest you forgo the benefits of ESLint? Does that sound fair to you?

**Comment by ljharb** [comment]
I’m not saying anything about an app. Every website should use a bundler, and of course my first website wasn’t using a bundler, because they didn’t exist then - but we’ve improved, and they exist now, and now they should be used. It’s never infeasible or improper to use a bundler to target a browser.

**Comment by adamlui** [comment]
The past you I was trying to get you to see was hacking a website with a simple script not building a website (maybe it wasn't you, that was me and countless others today getting into JS) so bundling makes no sense. But pretend even if it did, that's still not a reason to alienate JS users who just want to lint code not be lectured on build processes

**Comment by ljharb** [comment]
I’m very familiar with that model. It’s not a good one, it’s just an old one.

Again, this is a linter plugin, its job is to lecture you.

**Comment by adamlui** [comment]
> I’m very familiar with that model. It’s not a good one, it’s just an old one.

Why is it not good? Check out my script that does just that with over 100000 users that does not involve bundling and neccessitates importing resources as URLs https://github.com/adamlui/chatgpt-infinity/blob/main/greasemonkey/chatgpt-infinity.user.js

> Again, this is a linter plugin, its job is to lecture you.

Not on build proesses, did you look at the screenshot? ESLint lints code not build processes...

**Comment by adamlui** [comment]
Even if you think it _should_ be the job, the fact remains it loudly markets itself (as do every other linter) as lecturing code because by definition that's all it's supposed to do. So ESLint users would expect the same, so it is very unusual to alienate users for not adhering to what you subjectively think they should or should not do (i.e. don't write userscripts or URL imports, when countless people do and still require linting code not build procvesses)

**Comment by adamlui** [comment]
That's like if I disabled my popular script for Mac users because i go "Don't use Mac. It's not a good OS, just an old OS." (replace w/ "Windows" if you want). It is very unusual for me to lose 50,000 users just beause I don't like their OS (my script's job is to enhance their ChatGPT usability, not lecture them on OS)

**Comment by adamlui** [comment]
<3 Mac I used it since it was called Mac OS (v7.5) don't yell at me just trying to point out the absurdity in your justifications for shutting out people who don't care what you think they should use, they simply wrote code and are trying to ESLint

**Comment by adamlui** [comment]
I meant because you ridiculed my OS in another ticket so I don't want to incur your wraith for using Mac in an example, it was just an analogy

**Comment by adamlui** [comment]
I was up all night hacking so have to shower and sleep or else my long COVID I had for 4 years now makes me toss and turn from difficulty breathing but I promise to reply tomorrow to anything you wanna say so sweet dreams for now
